### PR TITLE
Update gate dictionary in `two_local.py`

### LINKED ISSUE #12002: Add `SXGate` to supported gates in `TwoLocal` circuits
### What should we add?

A common decomposition of single qubit gates is `Rz-Sx-Rz-Sx-Rz`, therefore it would be nice to support `SXGate`s in the `TwoLocal` circuit rotation gates.

https://github.com/Qiskit/qiskit/blob/230e91cb810de8f3911fa1958815fe8ed25c9980/qiskit/circuit/library/n_local/two_local.py#L276

### PR DESCRIPTION
**Pull request by amaloney**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.

### Summary

This commit removes the hard coded gate dictionary with the one generated by the method `get_standard_gate_name_mapping`.

### Details and comments

Resolves #12002

### CODE REVIEW

**Review by jakelishman — COMMENTED** [review]
This looks great, thanks!  Please could you add a bugfix or features_circuit note (whichever you think is more appropriate) just to comment that all of the standard-library gates are now supported as valid string gates?

**Comment by amaloney** [comment]
rebased against main as well as added a release note

File: releasenotes/notes/update-gate-dictionary-c0c017be67bb2f29.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+fixes:
+  - |
+    Fixes adding all supported gates in ``TwoLocal`` circuits. Refer to
+    `#12002 <https://github.com/Qiskit/qiskit/issues/12002>` for more
+    details.
+features_circuits:
+  - |
+    All of the gates in the ``standard_gates`` library are now supported
+    as valid sting gates for ``TwoLocal`` circuits.
```
Comment (atharva-satpute) [inline releasenotes/notes/update-gate-dictionary-c0c017be67bb2f29.yaml:10]: typo

Comment (atharva-satpute) [inline releasenotes/notes/update-gate-dictionary-c0c017be67bb2f29.yaml]:
```suggestion
    as valid string gates for ``TwoLocal`` circuits.
```

Comment (amaloney) [reply releasenotes/notes/update-gate-dictionary-c0c017be67bb2f29.yaml]: good catch

**Review by jakelishman — DISMISSED** [review]
Thanks for this! Sorry I dropped the re-review for quite a while there.

**Review by jakelishman — APPROVED** [review]
Just to explain - I pushed up a tweak to the release note since I think this is _either_ a feature or a bugfix but not both (I went "feature", since that seemed to be the way you'd gone preferentially as well) and to add a couple of additional Sphinx cross-references to it. Don't worry at all about that - we very frequently tweak release notes at the point of publishing a release anyway, just to make the "voice" a bit more consistent.

I also threw in a test case for the request in the issue you solved - it was my mistake I didn't think about that before, and I didn't want to send you off on another round when I already was super slow coming back to the review.

Thanks very much for the contribution!

**Pull request merged**